libraries/AP_Motors/AP_Motors_Class.cpp

```diff
@@ -18,20 +18,63 @@
 #include <SRV_Channel/SRV_Channel.h>
 #include <GCS_MAVLink/GCS.h>
 #include <AP_Notify/AP_Notify.h>
+#include <AP_Vehicle/AP_Vehicle_Type.h>
+
+// this attempts to capture old default defines, but is not fool proof as it cannot see the
+// old vehicle level defines that may have been added to defines.h,
+// but will work with defines from hwdef
+#ifndef PWM_RATE_DEFAULT
+    #if APM_BUILD_TYPE(APM_BUILD_ArduCopter) || APM_BUILD_TYPE(APM_BUILD_Blimp)
+        #ifdef RC_FAST_SPEED
+            #define PWM_RATE_DEFAULT RC_FAST_SPEED
+        #endif
+
+    #elif APM_BUILD_TYPE(APM_BUILD_Heli)
+        #define PWM_RATE_DEFAULT 125
+
+    #elif APM_BUILD_TYPE(APM_BUILD_ArduSub)
+        #ifdef RC_SPEED_DEFAULT
+            #define PWM_RATE_DEFAULT RC_SPEED_DEFAULT
+        #else
+            #define PWM_RATE_DEFAULT 200
+        #endif
+
+    #endif
+
+    #ifndef PWM_RATE_DEFAULT
+        #define PWM_RATE_DEFAULT 490
+    #endif
+#endif
+
+const AP_Param::GroupInfo AP_Motors::var_info[] = {
+
+    // @Param: PWM_RATE
+    // @DisplayName: ESC Update rate
+    // @Description: This is the rate in Hertz that your ESCs will receive updates
+    // @Units: Hz
+    // @Range: 50 490
```

Comment:
As a side note it seems that we sometimes set this to out-of-range values (see skyviper) but this is an existing issue and we don't need to fix it in this PR... just FYI.